server/raft.go

```diff
@@ -3761,6 +3761,10 @@ func (n *raft) readTermVote() (term uint64, voted string, err error) {
 	if err != nil {
 		return 0, noVote, err
 	}
+	if len(buf) < 8 {
```

Comment:
Also be good to understand how this could have ever happened.

Maybe run full test suite and add in check when raft group goes away to check for this condition.